### DISCUSSION TITLE
How to get list of all images from supabase bucket?

### DISCUSSION THREAD

**Discussion by nicitaacom** [supabase/supabase#19962] [Questions]
I use this code
```ts
  useEffect(() => {
    async function getListOfImages() {
      console.log(70, "debounced - ", debounced)
      const { data } = await supabaseClient.storage.from("ingredients").list(`${category}/${debounced}`)
      console.log(72, "data - ", data)
    }
    getListOfImages()
  }, [category, debounced])
```
 
it doesn't work and returns empty array

I want to get all list of images that contains S
![image](https://github.com/supabase/supabase/assets/39565703/96deee78-6a81-4fc4-a756-7101de0adbb9)

## Espected behavior
For example I type S and got all images and image names that includes S character

## Current behavior
I got emptly array

## How you may help
Please send docs where described how to do it
Send some minimal example where you do it
Send some minimal code needed to get expected behavior

**Comment by GaryAustin1** [comment]
Not at all clear to me what your call is doing.  The list command does have a search option but you are not using and seem to be setting three folder name. 

This show using search https://supabase.com/docs/reference/javascript/storage-from-list?example=search-files-in-a-bucket

**Comment by nicitaacom** [comment]
## Solution

```ts
  async function getAllImagesFromBucket() {
    const { data } = await supabaseClient.storage
      .from("ingredients")
      .list(`${category}`, { sortBy: { column: "name" } })

    const names = data?.map(bucketItem => bucketItem.name) || []
    setIngredientNames(names)
  }
```



then create your logic to set this items in some state and then if input value includes some item in string[] display this and highlight matched characters



```ts
const findMatchingItems = (names: string[]) => {
    if (watch("name") === "") {
      setMatchingItems([])
      return
    }
    const query = watch("name").toLowerCase()
    const matchingItems = names.filter(name => name.toLowerCase().includes(query))

    console.log(55, "setIsDropdown ture")
    setIsDropdown(true)
    setMatchingItems(matchingItems)
  }

  useEffect(() => {
    getAllImagesFromBucket()
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [category])

  useEffect(() => {
    findMatchingItems(ingredientNames)
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [watch("name"), ingredientNames])
  ```
  
  then create some logic for enter and arrowup and arrowdown and esc
  
  ### Anwering on question - you may get all images just by getting list of all images 
  ```ts
    const { data } = await supabaseClient.storage
      .from("ingredients")
      .list(`${category}`, { sortBy: { column: "name" } })

    const names = data?.map(bucketItem => bucketItem.name) || []
    setIngredientNames(names)
 ```
 
 and then with query to get image by name
 ```ts
   const { data } = supabaseClient.storage.from("ingredients").getPublicUrl(`${category}/${ingredientName}`)
   ```
   
   And then you set it in some state or do something with data
